src/App.tsx

```diff
@@ -29,6 +30,43 @@ function App() {
   const [error, setError] = useState<string | null>(null);
   const [loading, setLoading] = useState(true);
   const [dataHtml, setDataHtml] = useState("");
+  const [projectData, setProjectData] = useState<{
+    projectName: string;
+    artist: string;
+  } | null>(null);
+
+  useEffect(() => {
+    if (contractAddress === undefined || projectId === undefined) return;
+
+    const controller = new AbortController();
```

Comment:
could this just be a simple `isMounted` boolean variable (since viem doesn't integrate with controllers)?